api/pkg/cli/cmd/check_upgrade/check_uprade.go

```diff
@@ -0,0 +1,211 @@
+// Copyright © 2021 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package check_upgrade
+
+import (
+	"strings"
+	"text/template"
+
+	"github.com/spf13/cobra"
+	"github.com/tektoncd/hub/api/pkg/cli/app"
+	"github.com/tektoncd/hub/api/pkg/cli/formatter"
+	"github.com/tektoncd/hub/api/pkg/cli/hub"
+	"github.com/tektoncd/hub/api/pkg/cli/installer"
+	"github.com/tektoncd/hub/api/pkg/cli/kube"
+	"github.com/tektoncd/hub/api/pkg/cli/printer"
+)
+
+const upgradeTemplate = `{{decorate "underline bold" "Hub Tasks\n" }}
+{{- $len := len .HubResources }}{{- if ne $len 0 }}
+NAME	CURRENT_VERSION	NEWER_VERSION
+{{- range $el := .HubResources }}
+{{ $el.Name }}	{{ $el.CurrentVersion }}	{{ $el.LatestVersion }}
+{{- end }}
+{{- else }}
+No Tasks for Upgrade
+{{- end }}
+{{ $len := len .NonHubResources }}{{ if ne $len 0 }}
+{{ decorate "underline bold" "Skipped Tasks\n" }}
+NAME
+{{- range $el := .NonHubResources }}
+{{ icon "bullet" }}{{ $el }}
+{{- end }}
+{{- end }}
+`
+
+var (
+	funcMap = template.FuncMap{
+		"icon":     formatter.Icon,
+		"decorate": formatter.DecorateAttr,
+	}
+	tmpl = template.Must(template.New("Check Upgrade").Funcs(funcMap).Parse(upgradeTemplate))
+)
+
+type hubRes struct {
+	Name           string
+	CurrentVersion string
+	LatestVersion  string
+}
+
+type templateData struct {
+	NonHubResources []string
+	HubResources    []hubRes
+}
+
+const (
+	defaultCatalog = "tekton"
+	versionLabel   = "app.kubernetes.io/version"
+	hubLabel       = "hub.tekton.dev/catalog"
+)
+
+type options struct {
+	cli     app.CLI
+	from    string
+	kind    string
+	version string
+	args    []string
+	kc      kube.Config
+	cs      kube.ClientSet
+}
+
+var cmdExamples string = `
+Check for Upgrades of %S installed via Tekton Hub CLI:
+
+	tkn hub check-upgrades %s
+
+The above command will check for upgrades of %Ss installed via Tekton Hub CLI
+and will skip the %Ss which are not installed by Tekton Hub CLI.
+`
+
+func Command(cli app.CLI) *cobra.Command {
+	opts := &options{cli: cli}
+
+	cmd := &cobra.Command{
+		Use:   "check-upgrade",
+		Short: "Check for upgrades of resources if present",
+		Long:  ``,
+		Annotations: map[string]string{
+			"commandType": "main",
+		},
+		SilenceUsage: true,
+	}
+	cmd.AddCommand(
+		commandForKind("task", opts),
+	)
+
+	cmd.PersistentFlags().StringVar(&opts.from, "from", defaultCatalog, "Name of Catalog to which resource belongs.")
```

Comment:
are we using `from` flag?

Reply:
yes...in the API call we are passing the `opts.from`
```go
res := hubClient.GetResource(hub.ResourceOption{
			Name:            resource.GetName(),
			Catalog:         opts.from,
			Kind:            opts.kind,
			PipelineVersion: resInstaller.GetPipelineVersion(),
		})
```

Reply:
task installed through hub has catalog label right? can't we use that? if there are resources installed from multiple catalog then how would we check for upgrade then?